modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/ExtensionMessageSource.java

```diff
@@ -212,12 +212,14 @@ private void startSource(boolean restarting, RestartContext restartContext) thro
     Consumer<Throwable> onFailure;
     if (retryPolicyTemplate.isAsync()) {
       onSuccess = this::onReconnectionSuccessful;
-      onFailure = this::onReconnectionFailed;
+      onFailure = (t) -> {
+        dispatchExceptionNotification(t);
```

Comment:
in this case, is the dispatch method being called twice? once here, and another time in the first line of the onReconnectionFailed method.

Reply:
Yes, it is called twice on the async case. I'll fix it